python/ray/rllib/env/base_env.py

```diff
@@ -215,11 +227,14 @@ def _poll(self):
                 all_infos[eid] = data["info"]
                 if "off_policy_action" in data:
                     off_policy_actions[eid] = data["off_policy_action"]
-        return _with_dummy_agent_id(all_obs), \
-            _with_dummy_agent_id(all_rewards), \
-            _with_dummy_agent_id(all_dones, "__all__"), \
-            _with_dummy_agent_id(all_infos), \
-            _with_dummy_agent_id(off_policy_actions)
+        if self.multiagent:
+            return all_obs, all_rewards, all_dones, all_infos, off_policy_actions
```

Comment:
I think the way to fix your "reward key error" below is to inject 0.0 for any missing rewards here. Basically, to make sure the obs, rewards, infos, etc. dictionaries here have a consistent set of keys.